step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -784,6 +740,98 @@ public void cleanup() {
         }
     }
 
+    public static class DefaultExecutionTaskParameterHook implements AutomationPackageHook<ExecutiontTaskParameters> {
+
+        public DefaultExecutionTaskParameterHook() {
+        }
+
+        @Override
+        public void onAdditionalDataRead(String fieldName, List<?> yamlData, AutomationPackageContent targetContent, AutomationPackageReader reader) {
+           targetContent.getSchedules().addAll((Collection<? extends AutomationPackageSchedule>) yamlData);
+        }
+
+        @Override
+        public void onPrepareStaging(String fieldName, AutomationPackageContext apContext,
+                                     AutomationPackageContent apContent, List<?> objects,
+                                     AutomationPackage oldPackage, Staging targetStaging,
+                                     AutomationPackageManager manager) {
+            targetStaging.getTaskParameters().addAll(prepareExecutionTasksParamsStaging((List<AutomationPackageSchedule>) objects, apContext.getEnricher(), apContent, oldPackage, targetStaging.getPlans(), manager));
```

Comment:
if we want to use hooks completely for schedules we should not have targetStaging.getTaskParameters() but targetStaging.getAdditionalObjects().get(...)

Reply:
@david-stephan 

Implemented. Now everyting related to `schedules` in AP is moved to `additionalFields` and managed via hooks.